doc/source/general/compiling.rst

```diff
@@ -18,7 +18,33 @@ When you try to run your script, the first thing that kOS does is transform your
 
 This process of transforming your script into Machine Language, or "ML" is called "Compiling".
 
-The ":ref:`RUN`" and ":ref:`RUNPATH`" commands do this silently, without telling you. This is why you may have noticed the universe slightly stutter for a moment when you first run your program. Compiling is hard work for the universe to do.
+The :key:`RUN` and :func:`RUNPATH` commands do this silently, without telling
+you.  The :key:`COMPILE` command explicitly compiles the file and saves it for
+future use.
+
+.. note::
+    .. versionchanged:: 1.1.0
+        kOS now performs compilation on a parallel thread (thought to run in a
+        parallel universe).  This means that the universe no longer needs to
+        freeze in order to perform the compilation.  As with the execution of
+        other triggers, compiling from mainline code will not interupt or pause
+        triggers but compiling within a trigger will result in both the mainline
+        and trigger code pausing until the compilation is complete.
```

Comment:
I don't think the high-level effect of the threaded compile is that clear here - that from the kOS computer's point of view, it doesn't *look* multi-threaded.  The relevant high-level change here is that the compile takes time to finish, when before it happened instantly as far as the simulated universe was concerned and it froze the universe in order to accomplish this.  I think that fact should get top billing in this paragraph.  The fact that under the hood we used threading to do this is an implementation detail the script program never really sees.

Let me give it a go and then you can decide if you like yours better, or mine, or maybe a mix of the two:

```
        kOS no longer freezes the Kerbal Space Program universe when it compiles programs.  Now 
        only your kOS program waits for the compile to finish, rather than the entire KSP game.
        When you issue a RUN or RUNPATH command that requires the program to be compiled,
        the KSP universe will continue moving forward in time while the kOS CPU waits for the compile
        to complete.  The only effect this will have on programs you run is that if you run a program
        you have to be aware that it may be several game-seconds later before the program actually
        starts.  During that time the ship may have moved a bit, or fell a short distance, etc.

        (To programmers familiar with Unity: Under the hood, this change was
        accomplished by moving the compile to a separate thread so kOS's
        FixedUpdate can abort early and allow the rest of KSP's universe to
        keep moving.  kOS's FixedUpdate continues aborting early, not
        executing any of its CPU instructions, until the compile thread is done,
        at which point it resumes execution where it left off.)
```